lib/xnvme_dev.c

```diff
@@ -17,25 +17,17 @@ _zoned_geometry(struct xnvme_dev *dev)
 	struct xnvme_spec_znd_idfy_ns *zns = (void *)xnvme_dev_get_ns_css(dev);
 	struct xnvme_spec_znd_idfy_lbafe *lbafe = &zns->lbafe[nvm->flbas.format];
 	struct xnvme_geo *geo = &dev->geo;
-	uint64_t nzones;
-	int err;
 
 	if (!zns->lbafe[0].zsze) {
 		XNVME_DEBUG("FAILED: !zns.lbafe[0].zsze");
 		return -EINVAL;
 	}
 
-	err = xnvme_znd_stat(dev, XNVME_SPEC_ZND_CMD_MGMT_RECV_SF_ALL, &nzones);
```

Comment:
If this method is not used here, then I think it can just be removed

Reply:
The function ``xnvme_znd_stat()`` is convenient helper-function provided by the xNVMe in the public API for users of ZNS capable devices.